python/llm/src/bigdl/llm/ggml/transformers/model.py

```diff
@@ -0,0 +1,104 @@
+                  This should be a dir path that contains: weight bin, tokenizer config,
+                  tokenizer.model (required for llama) and added_tokens.json (if applied).
+                  For lora fine tuned model, the path should be pointed to a merged weight.
+               2. path for converted ggml binary checkpoint. The checkpoint should be converted by
+                  ``bigdl.llm.ggml.convert_model``.
+               3. a str for huggingface hub repo id.
+
+        :param model_family: the model family of the pretrained checkpoint.
+               Currently we support ``"llama"``, ``"bloom"``, ``"gptneox"``.
+        :param dtype: (optional) the data type for weight. Currently we only support ``"int4"``
+        :param cache_dir: (optional) this parameter will only be used when
+               ``pretrained_model_name_or_path`` is a hugginface checkpoint or hub repo id.
+               It indicates the saving path for the converted low precision model.
+        :param **kwargs: keyword arguments which will be passed to the model instance
+
+        :return: a model instance
+        """
+        invalidInputError(model_family in ['llama', 'gptneox', 'bloom'],
+                          "Now we only support model family: 'llama', 'gptneox', 'bloom', "
+                          "'{}' is not in the list.".format(model_family))
+        invalidInputError(dtype == 'int4',
+                          "Now we only support 'int4' as date type for weight")
+
+        # check whether pretrained_model_name_or_path exists.
+        # if not, it is likely that the user wants to pass in the repo id.
+        if not os.path.exists(pretrained_model_name_or_path):
+            try:
+                # download from huggingface based on repo id
+                pretrained_model_name_or_path = snapshot_download(
+                    repo_id=pretrained_model_name_or_path)
+            except Exception as e:
+                traceback.print_exc()
+                # if downloading fails, it could be the case that repo id is invalid,
+                # or the user pass in the wrong path for checkpoint
+                invalidInputError(False,
+                                  "Please input valid huggingface hub repo id, "
+                                  "or provide the valid path to huggingface / "
+                                  "ggml binary checkpoint, for pretrained_model_name_or_path")
+
+        ggml_model_path = pretrained_model_name_or_path
+        # check whether pretrained_model_name_or_path is a file.
+        # if not, it is likely that pretrained_model_name_or_path
+        # points to a huggingface checkpoint
+        if not os.path.isfile(pretrained_model_name_or_path):
+            # huggingface checkpoint
+            ggml_model_path = convert_model(input_path=pretrained_model_name_or_path,
+                                            output_path=cache_dir,
```

Comment:
`output_path=None` will be deprecated soon @rnwang04

Reply:
I will remove the default `None` value later after `convert_model` is changed

Reply:
Have changed `cache_dir` default value to `'./'`, this will work after `convert_model` supports folder as output_path